### ISSUE TITLE
Poor performance on custom dataset

### ISSUE THREAD

**Issue by blackmrb** [georghess/neurad-studio#26]
Hello, thanks for the great work firstly!

I reproduced pandaset based on your default configuration and it works great.

But the effect on my own data was very poor. I tried neurader and neuradest, but the effect was not good.

The data sensor configuration I am using now is an overhead hesai panda128, 4 cameras, the vehicle speed is about 30km/h, and the camera configuration is as follows

front_far（3840x2160），front_wide（3840x1630），

front_left（1936x1216），front_back（1936x1216）

The length of my data is 10 seconds and 100 frames in total. Except for not turning on add_miss_lidar, I used the default configuration. For the vertical and horizontal angles of the lidar, I referred to VLS128 and changed it to the parameters of hesai panda128.

GT view

![image](https://github.com/georghess/neurad-studio/assets/165770555/17437aca-172c-476a-a1f3-e9e96fe5c841)

render view(using neurader)
![image](https://github.com/georghess/neurad-studio/assets/165770555/7746cf82-f244-441f-9920-e144d4e74db8)

I tried neurader and neuradest, but the effect was not good. Here is the loss.
# loss

### Eval Loss
![image](https://github.com/georghess/neurad-studio/assets/165770555/4f75518b-cbaf-439e-a59e-fb07aa638518)
# Train loss
![image](https://github.com/georghess/neurad-studio/assets/165770555/aad5e699-f15c-4b63-8582-4efb95fd00e0)
![image](https://github.com/georghess/neurad-studio/assets/165770555/7445082d-3b94-43e1-8578-d0703f3a7c44)

Could you give me some advise to optimize? Many Thanks~

**Comment by blackmrb** [comment]
Hi, I train`neurad` method using default config on pandset 008 016 021 023 041 043, I want to test the lane shift performance. Because lane shift perfromance on my custom dataset is poor, I think I should test pandaset the performance firstly.
But the result is also not good. 
I set the lane shift distance to 2m left or right.  The top is the nvs image, the bottom is raw image. 
As you can see, the lane become very crooked or blurry, I don't konw what's the reason. Are there any hyperparameters that need to be adjusted?  
Hope your reply, thanks~
008
![image](https://github.com/georghess/neurad-studio/assets/165770555/c25e561e-3b42-4ee0-8830-6fbca1379012)
016
![image](https://github.com/georghess/neurad-studio/assets/165770555/e3092f60-d7e5-49d5-9c3a-36a53a016b1c)
041
![image](https://github.com/georghess/neurad-studio/assets/165770555/43c46c6d-1279-4c62-9c12-cce4be3e3084)

**Comment by blackmrb** [comment]
Thanks for the replay. 

For my custom dataset, I have 3D tracks for moving objects. The neurader and neuradest stay the same as neurad.

For the pandaset, I use 360 cameras, I used the default training cmd `python nerfstudio/scripts/train.py neurad --pipeline.datamanager.num_processes=0 pandaset-data --sequence $scene_id`.

I test several method, just like EmerNerf, StreetSurf, PVG, all these method's novel view is not good, but the original pose rendered view seems good.   It caused a lot of confusion for me. 

How do you think about what factors affect the performance of NVS? Maybe the 3d geometry is not good, and network is overfitted in the original perspective?



> Hi, As for your custom dataset, it's a bit hard to say why performance is not better without having access to the data. I assume you have 3D tracks for your custom data to handle the moving objects, yes? When you tried neurader or neuradest, did the results change or stay the same?
> 
> For the pandaset results: that the lanes become blurry and crooked to some degree is a know issue. However, for 016 it seems to be worse than what I'd expect. Is this when training with 360 cameras or only part of the cameras?

**Comment by atonderski** [comment]
Unfortunately difficulty in generalizing to completely novel perspectives is an inherent limitation of these NeRF-like techniques. NeRFs are great for interpolation between seen views, but not so great for extrapolation which is what we see here.

One can improve this by adding additional components such as stronger regularization/priors (e.g. road is flat), generative model supervision (e.g. ReconFusion), but we don't support any of this out of the box in this repository. 

If you develop anything in this direction, feel free to open a Pull Request!

(closing since this is expected behavior)
